app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+# continue
+class UsersController < ApplicationController
+ before_action :authenticate_user!, only: [:index, :update]
+
+  def index
+   if current_user.role.name == "admin"
+     @user = User.all
+     else
+      redirect_to root_path, alert: "Unauthorized access"
```

Comment:
`redirect_to root_path, alert: "You are not authorized for this action."`

Reply:
alert updated